src/bout++.cxx

```diff
@@ -141,6 +144,18 @@ int BoutInitialise(int &argc, char **&argv) {
   set_file = DEFAULT_SET;
   log_file = DEFAULT_LOG;
 
+#if BOUT_HAS_GETTEXT
+  // Setting the i18n environment
+  try {
+    std::locale::global(std::locale(""));
```

Comment:
Should we report the locale name after (and before?) this? 

Do we also need to `imbue` the locale on the stream outputs or is the global setting done earlier enough that it is setup before we create anything else that may output?

Reply:
The `global` call sets the default locale for all subsequent calls, and is done first so takes care of this. Unfortunately this results in a problem for the `fr_FR` locale (at least on my machines): Reading an input like "dy = 0.2" sets `dy` to zero. Changing to "0,2" didn't seem to work either, so not sure what's happening there. For now may need to just select which parts of the locale to use (mainly just LC_MESSAGES). Alternatively we could `imbue` the `C` locale when parsing numbers.